src/cmd/health.js

```diff
@@ -0,0 +1,113 @@
+/* Copyright (C) 2020 NooBaa */
+'use strict';
+
+const os_util = require('../util/os_utils');
+const { make_https_request } = require('../util/http_utils');
+const minimist = require('minimist');
+
+
+
+const HELP = `
+Help:
+
+    "nsfs" is a noobaa-core command runs a local S3 endpoint on top of a filesystem.
+    Each sub directory of the root filesystem represents an S3 bucket.
+    Health command will return the health status of deployed nsfs.
+`;
+
+const USAGE = `
+Usage:
+
+    node src/cmd/health [options...]
+`;
+
+
+const OPTIONS = `
+Options:
+
+    --nsfs_type <type>         (default nc)   Set the nsfs type for heath check.
+
+`;
+
+function print_usage() {
+    console.warn(HELP);
+    console.warn(USAGE.trimStart());
+    console.warn(OPTIONS.trimStart());
+    process.exit(1);
+}
+
+const HOSTNAME = "localhost";
+const PORT = 6443;
+const SERVICE = "nsfs";
+
+async function nc_nsfs_health() {
+    const service_status = await os_util.exec('systemctl show -p ActiveState --value ' + SERVICE, {
+      ignore_rc: true,
+      return_stdout: true,
+      trim_stdout: true,
+    });
+    const pid = await os_util.exec('systemctl show --property MainPID --value ' + SERVICE, {
+      ignore_rc: true,
+      return_stdout: true,
+      trim_stdout: true,
+    });
+    const path = '';
+    let status_code;
+    try {
+        const response = await make_https_request({ HOSTNAME, port: PORT, path, method: 'GET',
+                              rejectUnauthorized: false });
+        if (response) {
+          status_code = response.statusCode;
+        }
+    } catch (err) {
+      console.debug('Error while pinging endpoint host :' + HOSTNAME + ', port ' + PORT);
+    }
+    const memory_status = await os_util.exec('systemctl status ' + SERVICE + ' | grep Memory ', {
+      ignore_rc: true,
+      return_stdout: true,
+      trim_stdout: true,
+    });
+    let memory;
+    if (memory_status) {
+      memory = memory_status.split("Memory: ")[1].trim();
+    }
+    let service_health = "OK";
+    if (service_status !== "active" || pid === "0" || status_code !== 200) {
+      service_health = "NOTOK";
+    }
+    const helath = {
+      service_name: 'nsfs',
+      status: service_health,
+      memory: memory,
+      checks: {
+          service: {
+              service_status: service_status,
+              pid: pid,
+          },
+          endpoint: {
+              endpoint_response: status_code,
+          },
+      }
+  };
+  console.log(helath);
+}
+
+
+async function main(argv = minimist(process.argv.slice(2))) {
+  try {
+
+    if (argv.help || argv.h) return print_usage();
+
+    const nsfs_type = argv.nsfs_type || 'nc';
+    if (nsfs_type === 'nc') {
+      return nc_nsfs_health();
+    }
```

Comment:
please add here a health script not supported yet print or something similar for a nsfs_type != nc